test/serializer/optimized-functions/DeadObjectAssign4.js

```diff
@@ -0,0 +1,17 @@
+// Copies of _5:2
+// Why? _5 is the variable for Object.assign, and there should be
```

Comment:
Extremely fragile indeed. Assuming the goal is to get the number of Object.assign calls down, would something like `Copies of assign:1` be appropriate?

Reply:
There are no usages of Object.assign though, it's declared and referenced as `_5` outside the function once in the module.

Reply:
I looked at the output, and now I understand... Annoying.

Reply:
With --debugNames turned on, we are getting one step closer to having repeatedly mentioned "assign" in the code. However, another (useless) intermediate variable is declared because the function itself is made an argument to the build node. Don't do that, code gets better, and we can start counting :-)

Reply:
Maybe you can better address this in a follow up as it means changing how all the current temporals works for Object.assign. On another note, using a single reference to `Object.assign` , is far better than doing runtime property accesses to do the same thing, even if it does yield better output. Doing `_$0.assign()` as worse than doing `_$0(...)`, as it means having to lookup `assign` on `_$0` everytime.

Reply:
We have to find a better way to test this before the changes can land.

Reply:
I added `memoizeReference("Object.assign")`. We still need to reference variables that change between tests though :/